src/runtime/type.ts

```diff
@@ -1478,7 +1477,7 @@ export class TypeVariableInfo {
       return true;
     }
     if (this._canReadSubset) {
-      this._resolution = this._canReadSubset;
+      this.resolution = this._canReadSubset;
```

Comment:
this causes extra logic to run (e.g. type validity checking and a bunch of other stuff) - is that what you want?

Reply:
this was how the original code worked before i unintentionally changed it in: https://github.com/PolymerLabs/arcs/commit/7286581bf92836fd58a46526d979f28a316a3c04#diff-0518aa032d5fe87bc5390c9140916bb8R1481
this is causing the `_canReadSubset` and `_canWriteSuperset` not be set to null at resolution, which is inconsistent with the rest of the code.